bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java

```diff
@@ -862,7 +861,7 @@ private JsonArray getJsonArray(String[] values) {
             // we have to strip ending "/" here because Kodi returns a not valid path and filename
             // "fanart":"image://http%3a%2f%2fthetvdb.com%2fbanners%2ffanart%2foriginal%2f263365-31.jpg/"
             // "thumbnail":"image://http%3a%2f%2fthetvdb.com%2fbanners%2fepisodes%2f263365%2f5640869.jpg/"
-            String encodedURL = URLEncoder.encode(StringUtils.stripEnd(url, "/"), StandardCharsets.UTF_8.name());
+            String encodedURL = URLEncoder.encode(url.replaceAll("/$", ""), StandardCharsets.UTF_8.name());
```

Comment:
That is a pretty heavy replacement, perhaps make a small utility function instead?

Reply:
I wish I could use Java 11.

Do you like this proposal:
```
    private String stripEnd(String str, String suffix) {
        int end = str.length();
        if (end == 0 || suffix.isEmpty()) {
            return str;
        }
        while (end > 0 && suffix.indexOf(str.charAt(end - 1)) != -1) {
            end--;
        }
        return str.substring(0, end);
    }
```

Reply:
You could simplify it to only accept a single char instead of a string for the suffix.

Reply:
Done